turn_by_turn/madng.py

```diff
@@ -0,0 +1,81 @@
+"""
+MAD-NG
+------
+
+This module provides functions to read and write ``MAD-NG`` turn-by-turn measurement files. These files
+are in the **TFS** format.
+
+"""
+
+from __future__ import annotations
+
+import logging
+from pathlib import Path
+
+import pandas as pd
+import tfs
+
+from turn_by_turn.structures import TbtData, TransverseData
+
+LOGGER = logging.getLogger()
+
+
+def read_tbt(file_path: str | Path) -> TbtData:
+    LOGGER.info("Starting to read TBT data")
+    """
+    Reads turn-by-turn data from the ``MAD-NG`` **TFS** format file.
+
+    Args:
+        file_path (str | Path): path to the turn-by-turn measurement file.
+
+    Returns:
+        A ``TbTData`` object with the loaded data.
+    """
+    df = tfs.read(file_path)
+    LOGGER.info("Starting to read TBT data from dataframe")
+
+    nturns = int(df.iloc[-1].loc["turn"])
+    npart = int(df.iloc[-1].loc["id"])
+    LOGGER.info(f"Number of turns: {nturns}, Number of particles: {npart}")
+
+    # Get the unique BPMs and number of BPMs
+    bpms = df["name"].unique()
+    nbpms = len(bpms)
+
+    # Set the index to the particle ID
+    df.set_index(["id"], inplace=True)
+
+    matrices = []
+    for particle_id in range(npart):
```

Comment:
Since MAD-NG counts from 1, let's do the same and go with `range(1, npart+1)`. We avoid potential errors of having to add 1 to `particle_id` everywhere afterwards. Could use a reminder comment too :)